goas/atomic_asset_transfer/risc0_proofs/zone_merge/src/main.rs

```diff
@@ -0,0 +1,44 @@
+use cl::{merkle, nullifier::Nullifier, PtxRoot};
+use goas_proof_statements::zone_funds::MergePrivate;
+use ledger_proof_statements::death_constraint::DeathConstraintPublic;
+use risc0_zkvm::guest::env;
+
+fn main() {
+    let MergePrivate {
+        funds_note,
+        zone_note,
+        merge_event,
+        merge_event_state_path,
+        txs_root,
+        balances_root,
+    } = env::read();
+
+    let input_root = funds_note.input_root();
+    let output_root = zone_note.output_root();
+    let nf = Nullifier::new(funds_note.input.nf_sk, funds_note.input.nonce);
```

Comment:
This is equiv. to calling `InputWitness::nullifier()`
```suggestion
    let nf = funds_note.input.nullifier();
```